Homework/Task2_2(1)/MatrixClasses.cs

```diff
@@ -0,0 +1,105 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Task2_2_1_
+{
+    using System;
+    using System.Text;
+
+    public class DiagonalMatrix
+    {
+        private int[] diagonalElements;
+
+        public int Size { get; private set; }
+
+        public DiagonalMatrix(params int[] elements)
+        {
+            if (elements == null)
+            {
+                Size = 0;
+                diagonalElements = new int[0];
+            }
+            else
+            {
+                Size = elements.Length;
+                diagonalElements = new int[Size];
+                Array.Copy(elements, diagonalElements, Size);
+            }
+        }
+
+        public int this[int i, int j]
+        {
+            get
+            {
```

Comment:
Move index validation to a separate private method, and use t in lines 37 and 43

Reply:
Added IsIndexValid method